src/components/auth/RegistrationForm.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+  <form
+    class="registration-form"
+    @submit.prevent="register(user)"
+  >
+    <InputCustom
+      label="username"
+      v-model="user.username"
+      :required="true"
+      @input="clearMessage"
+    />
+    <InputCustom
+      label="e-mail"
+      type="email"
+      v-model="user.email"
+      :required="true"
+      @input="clearMessage"
+    />
+    <InputCustom
+      label="password"
+      type="password"
+      v-model="user.password"
+      :required="true"
+    />
+    <InputCustom
+      label="repeat password"
+      type="password"
+      v-model="passwordToCheck"
+      :required="true"
+    />
+    <div class="message-container">
+      <div
+        class="response-message"
+        v-if="responseMessage.length"
+      >
+        {{ responseMessage }}
+      </div>
+      <div
+        class="error-message"
+        v-if="errorMessage.length"
+      >
+        {{ errorMessage }}
+      </div>
+      <div
+        class="error-message"
+        v-if="arePasswordsEntered && !isPasswordConfirmed"
+      >
+        wrong password
+      </div>
+    </div>
+    <ButtonText
+      style-type="bordered"
+      type="submit"
+      text="register"
+      :disabled="arePasswordsEntered && !isPasswordConfirmed"
+    />
+  </form>
+</template>
+
+<script>
+import InputCustom from '@/components/formElements/InputCustom.vue';
+import ButtonText from '@/components/formElements/ButtonText.vue';
+import { mapActions } from 'vuex';
+
+export default {
+  components: {
+    InputCustom,
+    ButtonText,
+  },
+  data: () => ({
+    user: {
+      username: '',
+      email: '',
+      password: '',
+    },
+    passwordToCheck: '',
+    responseMessage: '',
+    errorMessage: '',
+  }),
+  computed: {
+    arePasswordsEntered() {
+      return Boolean(this.user.password.length && this.passwordToCheck.length);
+    },
+    isPasswordConfirmed() {
+      return this.user.password === this.passwordToCheck;
+    },
```

Comment:
Do I understand correct that we can see the error before we finish typing passwords?
If it is like that it's not a good UX because user is not correct in this case - it just doesn't finish writing.